src/hooks/__internal__/useModalManager/useModalManager.ts

```diff
@@ -80,17 +95,31 @@ const useModalManager = ({
       if (modalRegistered.current) {
         ModalManager.removeModal(ref, triggerRefocusOnClose);
 
+        if (lastFocusedElement.current) {
+          setTimeout(() => {
+            lastFocusedElement.current?.focus();
```

Comment:
I think this may be necessary since the call is made within an asynchronous function. Although it’s unlikely the ref will change during that short time, TypeScript may still consider it possible.

Reply:
Yeah, unfortunately I'm still getting a possibly null warning, to avoid it I'd have to do have an if statement asserting `lastFocusedElement.current` again within the `setTimeout` to avoid it

Reply:
No worries, it won't ever be null if it passes the guard as the only point that happens is in the callback the timeout is executing but I get why TS can't guarantee it when it computes the types